model/simulationtests/phase_change.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+require 'openstudio'
+require_relative 'lib/baseline_model'
+
+model = BaselineModel.new
+
+heat_balance_algorithm = model.getHeatBalanceAlgorithm
+heat_balance_algorithm.setAlgorithm('ConductionFiniteDifference')
+
+# make a 1 story, 100m X 50m, 5 zone core/perimeter building
+model.add_geometry({ 'length' => 100,
+                     'width' => 50,
+                     'num_floors' => 1,
+                     'floor_to_floor_height' => 4,
+                     'plenum_height' => 1,
+                     'perimeter_zone_depth' => 3 })
+
+# add windows at a 40% window-to-wall ratio
+model.add_windows({ 'wwr' => 0.4,
+                    'offset' => 1,
+                    'application_type' => 'Above Floor' })
+
+# add ASHRAE System type 01, PTAC, Residential
+model.add_hvac({ 'ashrae_sys_num' => '01' })
+
+# add thermostats
+model.add_thermostats({ 'heating_setpoint' => 24,
+                        'cooling_setpoint' => 28 })
+
+# assign constructions from a local library to the walls/windows/etc. in the model
+model.set_constructions
+
+# set whole building space type; simplified 90.1-2004 Large Office Whole Building
+model.set_space_type
+
+# add design days to the model (Chicago)
+model.add_design_days
+
+# adjust material that throws because of heat_balance_algorithm.setAlgorithm('ConductionFiniteDifference')
+model.getMaterials.each do |material|
+  if material.name.to_s.include?('Metal Decking')
+    # ** Severe  ** InitialInitHeatBalFiniteDiff: Found Material that is too thin and/or too highly conductive, material name = METAL DECKING
+    # **   ~~~   ** High conductivity Material layers are not well supported by Conduction Finite Difference, material conductivity = 45.006 [W/m-K]
+    # **   ~~~   ** Material thermal diffusivity = 1.401E-005 [m2/s]
+    # **   ~~~   ** Material with this thermal diffusivity should have thickness > 8.69672E-002 [m]
+    # **   ~~~   ** Material may be too thin to be modeled well, thickness = 1.50000E-003 [m]
+    # **   ~~~   ** Material with this thermal diffusivity should have thickness > 3.00000E-003 [m]
+    material.setThickness(0.09)
+  end
+end
+
+# assign material property phase change properties
+model.getSurfaces.each do |surface|
+  if surface.surfaceType.downcase == 'wall'
+
+    surface.construction.get.to_Construction.get.layers.each do |layer|
+      next unless layer.to_StandardOpaqueMaterial.is_initialized
+
+      opt_phase_change = layer.createMaterialPropertyPhaseChange
+      if opt_phase_change.is_initialized
```

Comment:
change to raise if !initialized

Reply:
But doesn't "createXXX" return false if a material property was already created for a given material?

Reply:
I suppose if you have materials that are shared between constructions that is true yes.